internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -341,6 +341,13 @@ type Reconciler struct {
 	record event.Recorder
 }
 
+// composedRendered is a wrapper around a composed resource that tracks whether
+// it was successfully rendered or not.
+type composedRendered struct {
+	composed resource.Composed
```

Comment:
```suggestion
	resource resource.Composed
```

Nit: I read `cd` as an abbreviation for "composed", so `cd.resource` makes more sense to me than `cd.composed`.